eternalcore-core/src/main/java/com/eternalcode/core/feature/randomteleport/RandomTeleportService.java

```diff
@@ -74,10 +78,26 @@ private CompletableFuture<Location> getSafeRandomLocation(World world, int attem
             return CompletableFuture.failedFuture(new RuntimeException("Cannot find safe location"));
         }
 
-        int radius = this.randomTeleportSettings.randomTeleportRadius();
+        int radius = 0;
+        if (this.randomTeleportSettings.randomTeleportType() == RandomTeleportType.STATIC_RADIUS) {
+            radius = this.randomTeleportSettings.randomTeleportRadius();
+        }
+
+        if (this.randomTeleportSettings.randomTeleportType() == RandomTeleportType.WORLD_BORDER_RADIUS) {
+            WorldBorder worldBorder = world.getWorldBorder();
+            radius = (int) worldBorder.getSize() / 2;
+        }
+
+        boolean noneWorld = this.locationsConfiguration.spawn.isNoneWorld();
+        Location spawnLocation = !noneWorld
```

Comment:
Tak sobie myślę że jak ktoś ma ustawiony staticBorder to warto sprawdzić czy wylosowane kordynaty są na pewno w tym borderze (WorldBorder#isInside), a jak nie są to podzielić WorldBorder#getSize na pół jak wyżej.